clang/lib/Sema/SemaExpr.cpp

```diff
@@ -62,7 +62,34 @@
 
 using namespace clang;
 using namespace sema;
-
+class EnumNameVisitor : public RecursiveASTVisitor<EnumNameVisitor> {
+private:
+  std::string EnumName;
```

Comment:
Why is EnumName not stored as StringRef?

Reply:
EnumName is stored as a std::string because StringRef cannot perform concatenation. We need std::string to construct and store the fully qualified enum name, which involves concatenating strings.